staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1231,11 +1314,30 @@ func (c *cacheWatcher) add(event *watchCacheEvent, timer *time.Timer) bool {
 
 	closeFunc := func() {
 		// This means that we couldn't send event to that watcher.
-		// Since we don't want to block on it infinitely,
-		// we simply terminate it.
+		// Since we don't want to block on it infinitely, we simply terminate it.
+
+		// if the bookmarkAfterResourceVersionReceived hasn't been received we can safely terminate the watcher.
```

Comment:
nit: If ... (start with a capital letter).
Also please make the comment lines shorter